### PR TITLE
[MXNET-30] Add label augmentation support in image recordio2

### PR DESCRIPTION
**Pull request by nihui**
## Description ##

pass the label data to the augmenter implementation so that the label could be changed when image transformation applied during augmentation.
It is useful for regression task such as landmark localization.

### Essentials ###
- [ ] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by nihui** [comment]
retest this please

**Comment by CodingCat** [comment]
Hi, the community has passed to vote about associating the code changes with JIRA (https://lists.apache.org/thread.html/ab22cf0e35f1bce2c3bf3bec2bc5b85a9583a3fe7fd56ba1bbade55f@%3Cdev.mxnet.apache.org%3E)

We have updated the guidelines for contributors in https://cwiki.apache.org/confluence/display/MXNET/Development+Process, please ensure that you have created a JIRA at https://issues.apache.org/jira/projects/MXNET/issues/ to describe your work in this pull request and include the JIRA title in your PR as [MXNET-xxxx] your title where MXNET-xxxx is the JIRA id

Thanks!

**Comment by nihui** [comment]
jenkins please retest changes

File: src/io/image_iter_common.h [hunk 1]
```diff
@@ -85,6 +85,14 @@ class ImageLabelMap {
     CHECK(it != idx2label_.end()) << "fail to find imagelabel for id " << imid;
     return mshadow::Tensor<cpu, 1>(it->second, mshadow::Shape1(label_width));
   }
+  /*! \brief find a label for corresponding index, return vector as copy */
+  inline std::vector<float> FindCopy(size_t imid) const {
```
Comment (piiswrong) [inline src/io/image_iter_common.h]: FindLabel

Comment (nihui) [reply src/io/image_iter_common.h]: FindCopy is borrowed from the similar existing code [1], so I use FindCopy for consistency.
But if you do think FindLabel is better, I could update the name, and may open another PR to update the existing FindCopy code.

please tell me if I need to do this.

[1] https://github.com/apache/incubator-mxnet/blob/0fe04e9d778ac9c55e933ff3f21c4ddf28a4a101/src/io/iter_image_det_recordio.cc#L116

Comment (nihui) [reply src/io/image_iter_common.h]: @piiswrong  hi sincerly, any further suggestions ?

**Pull request merged**